src/test/java/ibm/jceplus/junit/base/memstress/BaseTestMemStressAES.java

```diff
@@ -9,87 +9,44 @@
 
 import ibm.jceplus.junit.base.BaseTestCipher;
 import java.security.AlgorithmParameters;
-import java.security.spec.AlgorithmParameterSpec;
 import java.util.Arrays;
 import java.util.Random;
 import javax.crypto.Cipher;
 import javax.crypto.KeyGenerator;
 import javax.crypto.SecretKey;
 import javax.crypto.SecretKeyFactory;
-import javax.crypto.spec.IvParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
-import org.junit.Assume;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.Assert.assertTrue;
 
 public class BaseTestMemStressAES extends BaseTestCipher {
 
     Random r = new Random(5);
     static int iteration = 0;
     static final byte[] plainText1024 = new byte[1024];
     static final byte[] plainText16 = "12345678".getBytes();
-
-    //--------------------------------------------------------------------------
-    //
-    //
-    static boolean warmup = false;
     protected SecretKey key = null;
     protected AlgorithmParameters params = null;
     protected Cipher cp = null;
     protected boolean success = true;
     protected int specifiedKeySize = 0;
     static int numTimes = 100;
     boolean printheapstats = false;
-    int DEFAULT_KEYSIZE = 256;
     byte[] encodedKey = null;
     KeyGenerator keyGenerator = null;
     SecretKey keyFromIBMJCE = null;
 
-
-
-    //--------------------------------------------------------------------------
-    //
-    //
-    public BaseTestMemStressAES(String providerName) {
-        super(providerName);
-        this.specifiedKeySize = DEFAULT_KEYSIZE;
-        try {
-            if (warmup == false) {
-                warmup = true;
-                warmup();
-            }
-        } catch (Exception e) {
-        }
-    }
-
-    //--------------------------------------------------------------------------
-    //
-    //
-    public BaseTestMemStressAES(String providerName, int keySize) throws Exception {
-        super(providerName);
-        this.specifiedKeySize = keySize;
-
-        Assume.assumeTrue(javax.crypto.Cipher.getMaxAllowedKeyLength("AES") >= keySize);
-
-        try {
-            if (warmup == false) {
-                warmup = true;
-                warmup();
-            }
-        } catch (Exception e) {
-        }
-
-    }
-
-    //--------------------------------------------------------------------------
-    //
-    //
+    @BeforeEach
     public void setUp() throws Exception {
 
-        encodedKey = new byte[(specifiedKeySize > 0 ? specifiedKeySize : 128) / 8];
+        this.specifiedKeySize = getKeySize();
+        encodedKey = new byte[specifiedKeySize / 8];
         r.nextBytes(plainText1024);
 
         //key = new SecretKeySpec(encodedKey, 0, encodedKey.length, "AES");
-        keyGenerator = KeyGenerator.getInstance("AES", "OpenJCEPlus");
-        keyGenerator.init(DEFAULT_KEYSIZE);
+        keyGenerator = KeyGenerator.getInstance("AES", getProviderName());
+        keyGenerator.init(getKeySize());
```

Comment:
We could use `this.specifiedKeySize` here instead of calling the `getKeySize()` method again.

Reply:
Yes agreed. I used `this.specifiedKeySize` in this location to save one method call.